### ISSUE TITLE
Editing of the filtered deck options triggers rebuild for the selected deck (not the edited deck)

### ISSUE THREAD

**Issue by tico-tico** [ankidroid/Anki-Android#5091]
Assume you have a real deck and some filted decks, all filtered are empty now.

```
RealDeck
FilteredDeckA
FilteredDeckB
FilteredDeckC
FilteredDeckD
```

Select **FilteredDeckA** (tap on it), then long tap on **FilteredDeckB**, select *Deck options*, do some changes that triggers deck rebuilding. Expectation: **FilteredDeckB** rebuilt and have cards, actual result: **FilteredDeckA** rebuilt and have cards.

It's for 2.8.5 beta1, can't check 2.9 right now.

**Comment by mikehardy** [comment]
Very interesting - thanks for reporting this. I haven't reproduced it locally on either current stable or the alpha but I don't see anything missing in the steps to reproduce in order to give it a try - thanks for the clear instructions. 

If you do get a chance you could follow the instructions here to join the google group needed to get alpha versions automatically from the play store: https://docs.ankidroid.org/manual.html#betaTesting

I suspect the bug will still be present as I'm not aware of changes in the area but I could be wrong. I can say the alpha versions are currently working fine and we'll beta before long so they won't destabilize. They are also forwards and backwards compatible with 2.8 in case they fail for you for some reason

**Comment by mikehardy** [comment]
@timrae - actually this might be right in the area you're working on with #4999

**Comment by timrae** [comment]
I don't think this is related to #4999

**Comment by mikehardy** [comment]
Agreed, just in the area and for me I sometimes get quick wins that way

**Comment by tico-tico** [comment]
Ok, i just tried a new 2.9alpha50, cool, it doesn't crash on the first run any more as it used to.

Speaking of the original issue, it still exists. I made a short screencast to show how it looks IRL. First **ArtikelDE [a]** was selected and then i long-pressed **ArtikelDE [n]** and edited its filter config. In the end we have new cards in **ArtikelDE [a]**, and not in **ArtikelDE [n]** as expected.


[AnkiDroid-20181106_125644.zip](https://github.com/ankidroid/Anki-Android/files/2552664/AnkiDroid-20181106_125644.zip)

**Comment by mikehardy** [comment]
Bummer the problem still exists but thanks for checking. And yes we ha e fixed many crashes between 2.8 and 2.9! Hoping we haven't added any and more alpha testers helps us know that so thanks for trying it

**Comment by tico-tico** [comment]
A little excerpt from **Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/FilteredDeckOptions.java**.

```java
    private void closeDeckOptions() {
        if (mPrefChanged) {
            // Rebuild the filtered deck if a setting has changed
            mCol.getSched().rebuildDyn(mCol.getDecks().selected());
        }
        finish();
        ActivityTransitionAnimation.slide(this, ActivityTransitionAnimation.FADE);
    }
```

I believe **mCol.getDecks().selected()** means current selected deck, and since long-press doesn't change selection, selected deck is being rebuilt.

**Comment by mikehardy** [comment]
Probably safe to say this breaks reviewing for custom study because you can't rebuild the deck reliably? The concept of "current deck id" is the area just touched in alpha50 so if this isn't fixed by that then similar logic as was developed there needs to be deployed here I believe

**Comment by tico-tico** [comment]
> Probably safe to say this breaks reviewing for custom study because you can't rebuild the deck reliably?

Yes exactly, beside that if I change some deck **A** options I expect nothing will happen to some other deck named **B** because it was selected before my edits.

> The concept of "current deck id" is the area just touched in alpha50 so if this isn't fixed by that then similar logic as was developed there needs to be deployed here I believe

Just touching does change the selection but long-press doesn't. And we need a long-press to get the deck's context menu with the *Deck options* command.
Is it possible to change the currently selected deck on long-press too? It will fix the issue.

**Comment by WiliTest** [comment]
I have the same problem on 2.16alpha29: going into the settings of the filtered deck rebuild it (which is annoying)

**Comment by mikehardy** [comment]
@WiliTest can you try the most up to date code just in case it's fixed? We're blocked from releasing to the play store so you have to install directly: https://github.com/ankidroid/Anki-Android/releases - if it's still happening you might open a new issue to make sure it gets attention

**Comment by WiliTest** [comment]
Apologies for the delay, no the problem isn't fixed on the last version (AnkiDroid-2.16alpha55-armeabi-v7a.apk). I'm thus opening a new issue.
